gem/optimise.py

```diff
@@ -226,8 +226,7 @@ def delta_elimination(sum_indices, factors):
                        for f in factors if isinstance(f, Delta)
                        for index in (f.i, f.j) if index in sum_indices]
 
-    # Drop ones
-    return sum_indices, [e for e in factors if e != one]
+    return sum_indices, factors
```

Comment:
The result of `delta_elimination` is only used for `sum_factorise` which won't be negatively affected by extra one factors.